### ISSUE TITLE
Support AWS Aurora MySQL as application database

### ISSUE THREAD

**Issue by imjustd** [metabase/metabase#10709]
**Describe the bug**
I can't start Metabase v0.33.0 with the JAR file download from http://downloads.metabase.com/v0.33.0/metabase.jar and connect to Amazon Aurora MySQL v5.6.10a from Ubuntu 19.04 with the command `sudo java -DMB_DB_CONNECTION_URI="mysql://<host>:3306/<database>?user=<user>&password=<password>" -Duser.timezone=UTC -jar metabase.jar`.

**Logs**
This are the logs I get on the terminal:
```
08-24 07:04:20 DEBUG plugins.classloader :: Using NEWLY CREATED classloader as shared context classloader: clojure.lang.DynamicClassLoader@538cd0f2
08-24 07:04:20 DEBUG plugins.classloader :: Setting current thread context classloader to shared classloader clojure.lang.DynamicClassLoader@538cd0f2...
08-24 07:04:20 INFO metabase.util :: Loading Metabase...
08-24 07:04:20 INFO metabase.util :: Maximum memory available to JVM: 7.0 GB
08-24 07:04:21 INFO util.encryption :: Saved credentials encryption is DISABLED for this Metabase instance. 🔓 
 For more information, see https://metabase.com/docs/latest/operations-guide/encrypting-database-details-at-rest.html
08-24 07:04:22 INFO metabase.driver :: Registered abstract driver :sql  🚚
08-24 07:04:24 INFO metabase.core :: Starting Metabase in STANDALONE mode
08-24 07:04:24 INFO metabase.server :: Launching Embedded Jetty Webserver with config: 
 {:port 3000}

08-24 07:04:24 INFO metabase.core :: Starting Metabase version v0.33.0 (06229af release-0.33.x) ...
08-24 07:04:24 INFO metabase.core :: System timezone is 'UTC' ...
08-24 07:04:24 INFO metabase.plugins :: Loading plugins in /home/dimitar/Desktop/test/plugins...
08-24 07:04:25 INFO plugins.dependencies :: Plugin 'Metabase BigQuery Driver' depends on plugin 'Metabase Google Drivers Shared Dependencies'
08-24 07:04:25 INFO plugins.dependencies :: Metabase BigQuery Driver dependency {:plugin Metabase Google Drivers Shared Dependencies} satisfied? false
08-24 07:04:25 INFO plugins.dependencies :: Plugins with unsatisfied deps: ["Metabase BigQuery Driver"]
08-24 07:04:25 INFO plugins.lazy-loaded-driver :: Registering lazy loading driver :snowflake...
08-24 07:04:25 INFO metabase.driver :: Registered abstract driver :sql-jdbc (parents: [:sql]) 🚚
Load driver :sql-jdbc took 26.4 ms
08-24 07:04:25 INFO metabase.driver :: Registered driver :snowflake (parents: [:sql-jdbc]) 🚚
08-24 07:04:25 INFO plugins.dependencies :: Metabase cannot initialize plugin Metabase Vertica Driver due to required dependencies. Metabase requires the Vertica JDBC driver in order to connect to Vertica databases, but we can't ship it as part of Metabase due to licensing restrictions. See https://metabase.com/docs/latest/administration-guide/databases/vertica.html for more details.

08-24 07:04:25 INFO plugins.dependencies :: Metabase Vertica Driver dependency {:class com.vertica.jdbc.Driver} satisfied? false
08-24 07:04:25 INFO plugins.dependencies :: Plugins with unsatisfied deps: ["Metabase Vertica Driver" "Metabase BigQuery Driver"]
08-24 07:04:25 INFO plugins.dependencies :: Plugin 'Metabase Google Analytics Driver' depends on plugin 'Metabase Google Drivers Shared Dependencies'
08-24 07:04:25 INFO plugins.dependencies :: Metabase Google Analytics Driver dependency {:plugin Metabase Google Drivers Shared Dependencies} satisfied? false
08-24 07:04:25 INFO plugins.dependencies :: Plugins with unsatisfied deps: ["Metabase Vertica Driver" "Metabase BigQuery Driver" "Metabase Google Analytics Driver"]
08-24 07:04:25 INFO plugins.lazy-loaded-driver :: Registering lazy loading driver :sqlite...
08-24 07:04:25 INFO metabase.driver :: Registered driver :sqlite (parents: [:sql-jdbc]) 🚚
08-24 07:04:25 INFO plugins.lazy-loaded-driver :: Registering lazy loading driver :redshift...
08-24 07:04:25 INFO metabase.driver :: Registered driver :postgres (parents: [:sql-jdbc]) 🚚
Load driver :postgres took 10.6 ms
08-24 07:04:25 INFO metabase.driver :: Registered driver :redshift (parents: [:postgres]) 🚚
08-24 07:04:25 INFO plugins.dependencies :: Metabase cannot initialize plugin Metabase Oracle Driver due to required dependencies. Metabase requires the Oracle JDBC driver in order to connect to Oracle databases, but we can't ship it as part of Metabase due to licensing restrictions. See https://metabase.com/docs/latest/administration-guide/databases/oracle.html for more details.

08-24 07:04:25 INFO plugins.dependencies :: Metabase Oracle Driver dependency {:class oracle.jdbc.OracleDriver} satisfied? false
08-24 07:04:25 INFO plugins.dependencies :: Plugins with unsatisfied deps: ["Metabase Vertica Driver" "Metabase BigQuery Driver" "Metabase Oracle Driver" "Metabase Google Analytics Driver"]
08-24 07:04:25 INFO plugins.lazy-loaded-driver :: Registering lazy loading driver :hive-like...
08-24 07:04:25 INFO metabase.driver :: Registered abstract driver :hive-like (parents: [:sql-jdbc]) 🚚
08-24 07:04:25 INFO plugins.lazy-loaded-driver :: Registering lazy loading driver :sparksql...
08-24 07:04:25 INFO metabase.driver :: Registered driver :sparksql (parents: [:hive-like]) 🚚
08-24 07:04:25 INFO plugins.lazy-loaded-driver :: Registering lazy loading driver :mongo...
08-24 07:04:25 INFO metabase.driver :: Registered driver :mongo  🚚
08-24 07:04:25 INFO plugins.lazy-loaded-driver :: Registering lazy loading driver :sqlserver...
08-24 07:04:25 INFO metabase.driver :: Registered driver :sqlserver (parents: [:sql-jdbc]) 🚚
08-24 07:04:25 INFO plugins.lazy-loaded-driver :: Registering lazy loading driver :google...
08-24 07:04:25 INFO metabase.driver :: Registered abstract driver :google  🚚
08-24 07:04:25 INFO plugins.dependencies :: Metabase BigQuery Driver dependency {:plugin Metabase Google Drivers Shared Dependencies} satisfied? true
08-24 07:04:25 INFO plugins.dependencies :: Metabase Google Analytics Driver dependency {:plugin Metabase Google Drivers Shared Dependencies} satisfied? true
08-24 07:04:25 DEBUG plugins.initialize :: Dependencies satisfied; these plugins will now be loaded: ["Metabase Google Analytics Driver" "Metabase BigQuery Driver"]
08-24 07:04:25 INFO plugins.lazy-loaded-driver :: Registering lazy loading driver :googleanalytics...
08-24 07:04:25 INFO metabase.driver :: Registered driver :googleanalytics (parents: [:google]) 🚚
08-24 07:04:25 INFO plugins.lazy-loaded-driver :: Registering lazy loading driver :bigquery...
08-24 07:04:25 INFO metabase.driver :: Registered driver :bigquery (parents: [:sql :google]) 🚚
08-24 07:04:25 INFO plugins.lazy-loaded-driver :: Registering lazy loading driver :druid...
08-24 07:04:25 INFO metabase.driver :: Registered driver :druid  🚚
08-24 07:04:25 INFO plugins.lazy-loaded-driver :: Registering lazy loading driver :presto...
08-24 07:04:25 INFO metabase.driver :: Registered driver :presto (parents: [:sql]) 🚚
08-24 07:04:25 INFO metabase.driver :: Registered driver :h2 (parents: [:sql-jdbc]) 🚚
Load driver :h2 took 8.8 ms
08-24 07:04:25 INFO metabase.driver :: Registered driver :mysql (parents: [:sql-jdbc]) 🚚
Load driver :mysql took 10.8 ms
08-24 07:04:25 INFO metabase.core :: Setting up and migrating Metabase DB. Please sit tight, this may take a minute...
08-24 07:04:25 INFO metabase.db :: Verifying mysql Database Connection ...
08-24 07:04:25 INFO metabase.driver :: Initializing driver :sql...
08-24 07:04:25 INFO metabase.driver :: Initializing driver :sql-jdbc...
08-24 07:04:25 INFO metabase.driver :: Initializing driver :mysql...
08-24 07:04:26 INFO metabase.db :: Verify Database Connection ...  ✅
08-24 07:04:26 INFO metabase.db :: Running Database Migrations...
```

**To Reproduce**
1. Download Metabase v0.33.0 from http://downloads.metabase.com/v0.33.0/metabase.jar
2. Setup Amazon RDS Aurora MySQL database v5.6.10a with charset `utf8` and collation `utf8_general_ci`
3. From Ubuntu 19.04 try to run `sudo java -DMB_DB_CONNECTION_URI="mysql://<host>:3306/<database>?user=<user>&password=<password>" -Duser.timezone=UTC -jar metabase.jar` (replace variables between <> with the Aurora MySQL variables)
4. View terminal logs, they should stop at the `INFO metabase.db :: Running Database Migrations...`

**Expected behavior**
The expected behavior is to finish the MySQL migrations without getting stuck and set up the MySQL tables on Amazon Aurora. (Metabase v0.31.2 does this without issues)

**Screenshots**
Logs screenshot on Ubuntu 19.10"
![image](https://user-images.githubusercontent.com/590260/63633922-5eb34100-c650-11e9-8784-46bb28a85cd6.png)

**Information about your Metabase Installation:**
- Your browser and the version: Chrome 76.0.3809.100 (Official Build) (64-bit)

- Your operating system: Ubuntu Linux 19.10 (fully updated with latest packages)

- Your databases: Amazon Aurora MySQL v5.6.10a

- Metabase version: v.0.33.0

- Metabase hosting environment: Locally started server with the JAR file (Ubuntu 19.04)

- Metabase internal database: H2 (not used)

**Severity**
How severe an issue is this bug to you? Is this annoying, blocking some users, blocking an upgrade or blocking your usage of Metabase entirely?
- I consider as a very severe bug since I can't use the latest Metabase and I'm forced to use v0.31.2 due to this issue. So not updates are applicable beyond v0.31.2 for me.

Note: the more honest and specific you are here the more we will take you seriously. 

**Additional context**
Note that Metabase v0.33.0 works without issues with its embedded H2 database, the issue is with the Amazon Aurora MySQL database.

The Aurora MySQL database I use is empty with charset `utf8` and collation `utf8_general_ci`.

*The older version of Metabase v0.31.2 works without any issues with the same MySQL setup on Amazon Aurora, but version v.0.32+ are not working at all.*

:arrow_down: Please click the :+1: reaction instead of leaving a `+1` or `update?` comment

**Comment by eugenedong** [comment]
+1, I'm facing the exact same issue. 

Running on AWS Aurora MySQL and is stuck on 0.31.2 as well because of the migration issue. 

Tried `migrate release-locks` and `migrate force` without success.

**Comment by flyingmachine** [comment]
Can you share what version of Java you're using?

**Comment by eugenedong** [comment]
> @imjustd and @eugenedong, thank you very much for providing this info. Can you share what version of Java you're using?

Apologies for the delay, here's the Java version output: 

```
$ java -version
openjdk version "1.8.0_181"
OpenJDK Runtime Environment (IcedTea 3.9.0) (Alpine 8.181.13-r0)
OpenJDK 64-Bit Server VM (build 25.181-b13, mixed mode)
```

**Comment by flyingmachine** [comment]
I've tried this on my local machine and on an Ubuntu VM running 19.10 and open-jd-8. `java -version` returns

openjdk version "1.8.0_232"
OpenJDK Runtime Environment (build 1.8.0_232-8u232-b09-0ubuntu1.1-b09)
OpenJDK 64-Bit Server VM (build 25.232-b09, mixed mode)

I tried to repro by both:

1. Starting 0.33.0 on a fresh install
2. Starting 0.31.2 on a fresh install, then starting on 0.33.0

So far, I haven't been able to repro this, but am sharing all my notes for future reference.

- The logs end at `Running Database Migrations...` The next log line should be [`Setting up Liquibase`](https://github.com/metabase/metabase/blob/v0.31.2/src/metabase/db.clj#L294). The fact that that wasn't printed means that MB didn't even attempt the actual migration
- I checked git history and as far as I can tell none of the code between `Running Database Migrations...` and `Setting up Liquibase` was changed between 0.31.2 and 0.33.0
- No migrations have been added since 0.30.0 (but again, the code didn't even get to the point of trying to run a migration)

It seems like the only possible lines that could have caused this freeze are

```clojure
(jdbc/db-set-rollback-only! conn)
;; Disable auto-commit. This should already be off but set it just to be safe
(.setAutoCommit (jdbc/get-connection conn) false)
```

However, as far as I could tell these lines (and the entire graph of functions they call) hadn't changed.

@imjustd and @eugenedong, are you still having this problem?

A couple thoughts on next steps:
- Add more logging to narrow in on what line is causing this
- Add a connection debug util if we don't already have one

**Comment by flyingmachine** [comment]
I'm going to close this for now because I couldn't reproduce it, but please feel free to comment or re-open if the issue persists.

I've created #11757 to aid in debugging issues like this in the future.

**Comment by Bukashk0zzz** [comment]
Has the same issue. And also has Aurora MySQL.
Wait when the master becomes release and made a test with MB_NS_TRACE="metabase.db"

**Comment by Bukashk0zzz** [comment]
Try run with 
image: metabase/metabase-head:release-0.34.x and env MB_NS_TRACE="metabase.db"

get in logs:
<img width="1678" alt="Screenshot 2020-01-29 at 11 01 22" src="https://user-images.githubusercontent.com/1908342/73342170-b18bc700-4286-11ea-9adc-a489e360f938.png">

**Comment by oliveirafilipe** [comment]
Hi @imjustd did you managed to get it working?

I'm facing the same issue while trying upgrade from `metabase/metabase:v0.31.2` to `metabase/metabase:v0.36.0`
My database is also an Aurora Mysql `5.7.mysql_aurora.2.08.1` and the version v0.31.2 connects without any issues.

**Comment by Bukashk0zzz** [comment]
I just stack on version v0.31.1, and that's it)
I did not found critical features for me in new releases so I decided not to waste time on this.

If you find solution, please add it to this ticket.

<img width="344" alt="Screenshot 2020-07-29 at 18 56 15" src="https://user-images.githubusercontent.com/1908342/88822945-2fe10b80-d1cd-11ea-8bab-9de98485999a.png">

**Comment by flamber** [comment]
Even Aurora Postgres is also behaving differently than expected, when trying to use as application database - #13014

**Comment by oliveirafilipe** [comment]
```
metabase_1  | 07-29 16:06:39 INFO driver.impl :: Initializing driver :mysql...
metabase_1  | TRACE t537: | | | | | (metabase.db/jdbc-spec {:type :mysql, :user "my-user", :password "MyPassHasExclamation!", :host "my-omitted-host.com", :port "3306", :dbname "metabase"})
metabase_1  | TRACE t537: | | | | | => {:classname "org.mariadb.jdbc.Driver", :subprotocol "mysql", :subname "//my-omitted-host.com:3306/metabase", :type :mysql, :user "my-user", :password "MyPassHasExclamation!", :dbname "metabase"}
metabase_1  | TRACE t538: | | | | | (metabase.db/db-is-setup?)
metabase_1  | TRACE t538: | | | | | => false
metabase_1  | 07-29 16:06:42 INFO metabase.db :: Successfully verified MySQL 5.7.12 application database connection. ✅
metabase_1  | TRACE t532: | | | | => nil
metabase_1  | TRACE t531: | | | => nil
metabase_1  | TRACE t539: | | | (metabase.db/run-schema-migrations! true {:type :mysql, :user "my-user", :password "MyPassHasExclamation!", :host "my-omitted-host.com", :port "3306", :dbname "metabase"})
metabase_1  | TRACE t540: | | | | (metabase.db/db-is-setup?)
metabase_1  | TRACE t540: | | | | => false
metabase_1  | 07-29 16:06:42 INFO metabase.db :: Running Database Migrations...
metabase_1  | TRACE t541: | | | | (metabase.db/jdbc-spec {:type :mysql, :user "my-user", :password "MyPassHasExclamation!", :host "my-omitted-host.com", :port "3306", :dbname "metabase"})
metabase_1  | TRACE t541: | | | | => {:classname "org.mariadb.jdbc.Driver", :subprotocol "mysql", :subname "//my-omitted-host.com:3306/metabase", :type :mysql, :user "my-user", :password "MyPassHasExclamation!", :dbname "metabase"}
metabase_1  | TRACE t542: | | | | (metabase.db/migrate! {:classname "org.mariadb.jdbc.Driver", :subprotocol "mysql", :subname "//my-omitted-host.com:3306/metabase", :type :mysql, :user "my-user", :password "MyPassHasExclamation!", :dbname "metabase"} :up)
```

This is the output from the initializing process, as I said for `metabase:v0.36.0`
Tried use both `MB_DB_CONNECTION_URI` and `MB_DB_*` without success.

**Comment by imjustd** [comment]
> Hi @imjustd did you manage to get it working?
> 
> I'm facing the same issue while trying upgrade from `metabase/metabase:v0.31.2` to `metabase/metabase:v0.36.0`
> My database is also an Aurora Mysql `5.7.mysql_aurora.2.08.1` and the version v0.31.2 connects without any issues.

No, I just stick with Metabase v0.31.2 for now.

**Comment by camsaul** [comment]
We only support MySQL 5.7+

**Comment by rlotun** [comment]
The last customer comment was from MySQL 5.7 and they can't upgrade to 36. Seems like something we should investigate. cc/ @camsaul @flamber

**Comment by imjustd** [comment]
We're going to migrate to MySQL 8.0, is this supported by Metabase or should we stick to MySQL 5.7?

**Comment by paoliniluis** [comment]
@imjustd yes it is

**Comment by imjustd** [comment]
Any specific Metabase version we need to update to, for MySQL 8.0, or also old Metabase versions support it?

**Comment by flamber** [comment]
@imjustd You should be able to use any Metabase version the past couple of years with MySQL 8:
https://www.metabase.com/docs/latest/operations-guide/configuring-application-database.html#mysql-or-mariadb
Though you might need to use the old authentication plugin:
https://www.metabase.com/docs/latest/administration-guide/databases/mysql.html

**Comment by imjustd** [comment]
@flamber Ok, thanks!

**Comment by paoliniluis** [comment]
Just confirmed that Metabase latest (at this time 39.4) works without issues with 5.7.mysql_aurora.2.10.0. The only additional step I had to do to connect Metabase was to manually create the database where Metabase will work on (in Aurora PostgreSQL you can do this step in the creation wizard), but once that's finished, everything worked alright

**Comment by flamber** [comment]
This was likely fixed in 0.38.0 (see #14678), where Metabase got stuck on the application database connection after it had verified the connection. This was because that part of the code was handled differently.
Metabase still requires MySQL 5.7.7 at a minimum (MySQL 5.6 is not supported): https://www.metabase.com/docs/latest/operations-guide/configuring-application-database.html#mysql-or-mariadb
